src/reanimated2/animation/decay/utils.ts

```diff
@@ -26,12 +21,48 @@ export interface DecayAnimation extends Animation<DecayAnimation> {
   current: AnimatableValue;
 }
 
-export interface DefaultDecayConfig {
-  deceleration: number;
-  velocityFactor: number;
-  clamp?: number[];
-  velocity: number;
+export interface InnerDecayAnimation
+  extends Omit<DecayAnimation, 'current'>,
+    AnimationObject {
+  current: number;
+  springActive?: boolean;
+}
+
+export type DecayConfig = {
+  deceleration?: number;
+  velocityFactor?: number;
+  velocity?: number;
   reduceMotion?: ReduceMotion;
-  rubberBandEffect?: boolean;
-  rubberBandFactor: number;
+} & (
+  | {
+      rubberBandEffect?: false;
+      clamp?: [min: number, max: number];
+    }
+  | {
+      rubberBandEffect: true;
+      clamp: [min: number, max: number];
+      rubberBandFactor?: number;
+    }
+);
+
+export type DefaultDecayConfig = RequiredKeys<
+  DecayConfig,
+  'deceleration' | 'velocityFactor' | 'velocity'
+> & { rubberBandFactor: number };
+
+// If user wants to use rubber band decay animation we have to make sure he has provided clamp
+export type RubberBandDecayConfig = RequiredKeys<
+  DefaultDecayConfig,
+  'clamp'
+> & { rubberBandEffect: true };
+
+export function isValidRubberBandConfig(
```

Comment:
I would prefer `isValidRubberBandConfig` to be declared just below `RubberBandConfig`, since Assertion Functions may be quite dangerous when left unattended 👀